migrations/20191129034013-initial_table_creation.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.createTable('movies', {
+      id: { type: Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
+      title: { type: Sequelize.STRING },
+
+      releaseDate: { type: Sequelize.DATEONLY },
+      rated: { type: Sequelize.STRING },
+      runTime: { type: Sequelize.INTEGER },
+
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP') },
+      deletedAt: { type: Sequelize.DATE }
+    })
+
+    await queryInterface.createTable('directors', {
+      id: { type: Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
+      name: { type: Sequelize.STRING },
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP') },
+      deletedAt: { type: Sequelize.DATE }
+    })
+
+    await queryInterface.createTable('genres', {
+      id: { type: Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
+      name: { type: Sequelize.STRING },
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP') },
+      deletedAt: { type: Sequelize.DATE }
+    })
+
+    await queryInterface.createTable('movie_director', {
+      id: { type: Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
+      movieId: { type: Sequelize.INTEGER, reference: { model: 'movie', key: 'id' } },
+      directorId: { type: Sequelize.INTEGER, reference: { model: 'director', key: 'id' } },
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP') },
+      deletedAt: { type: Sequelize.DATE }
+    })
+
+    return queryInterface.createTable('movie_genre', {
+      id: { type: Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
```

Comment:
Linking tables don't get their own ID, instead the movieId and directorId should be a compound key.